packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -130,6 +130,54 @@ function doesImmediatelyReturnFunctionExpression({
   );
 }
 
+/**
+ * Checks if a function belongs to:
+ * ```
+ * () => () => ...
+ * () => function () { ... }
+ * () => { return () => ... }
+ * () => { return function () { ... } }
+ * function fn() { return () => ... }
+ * function fn() { return function() { ... } }
+ * ```
+ */
+function doesImmediatelyReturnClassExpression({ body }: FunctionNode): boolean {
```

Comment:
@JoshuaKGoldberg Yes, my initial approach tried to be a bit more ambitious and recursively search for `return` statements in the function, but couldn't find a way to resolve the indentifiers so I aimed my draft a little lower. I did try to use `context.getScope()|`and it let me make some progress on `explicit-function-return-type`, but `explicit-module-boundary-types` seems to retrieve the global scope instead of the scope of the function definition... Is there any way to retrieve the scope of a specific node? And how about some util for searching all nested `return` statements of a body? If not, is there any documentation I could read about what the nodes within a body might be? I tried searching other rules that required a similar analysis but the code is quite vast, so any hints are welcome :)

Reply:
You know, that's a great question. I'm actually not too familiar with the scope manager myself, and the docs on https://eslint.org/docs/latest/developer-guide/scope-manager-interface aren't answering questions on what it can do around returns. Not super encouraging there. @armano2 or @bradzacher or @JamesHenry might know more.

Reply:
@nscarcella does this answer your questions? Or, is there more info that we can give to be helpful?